src/joystick/windows/SDL_xinputjoystick.c

```diff
@@ -114,116 +113,25 @@ static const char *GetXInputName(const Uint8 userid, BYTE SubType)
+        capabilities.VendorId = USB_VENDOR_MICROSOFT;
+        capabilities.ProductId = USB_PRODUCT_XBOX360_XUSB_CONTROLLER;
     }
 
-    for (i = 0; i < device_count; i++) {
-        RID_DEVICE_INFO rdi;
-        char devName[MAX_PATH] = { 0 };
-        UINT rdiSize = sizeof(rdi);
-        UINT nameSize = SDL_arraysize(devName);
-
-        rdi.cbSize = sizeof(rdi);
-        if (devices[i].dwType == RIM_TYPEHID &&
-            GetRawInputDeviceInfoA(devices[i].hDevice, RIDI_DEVICEINFO, &rdi, &rdiSize) != (UINT)-1 &&
-            GetRawInputDeviceInfoA(devices[i].hDevice, RIDI_DEVICENAME, devName, &nameSize) != (UINT)-1) {
-#ifdef DEBUG_JOYSTICK
-            SDL_Log("Raw input device: VID = 0x%x, PID = 0x%x, %s\n", rdi.hid.dwVendorId, rdi.hid.dwProductId, devName);
-#endif
-            if (SDL_strstr(devName, "IG_") != NULL) {
-                SDL_bool found = SDL_FALSE;
-                for (j = 0; j < SDL_arraysize(s_arrXInputDevicePath); ++j) {
-                    if (!s_arrXInputDevicePath[j]) {
-                        continue;
-                    }
-                    if (SDL_strcmp(devName, s_arrXInputDevicePath[j]) == 0) {
-                        found = SDL_TRUE;
-                        break;
-                    }
-                }
-                if (found) {
-                    /* We already have this device in our XInput device list */
-                    continue;
-                }
-
-                /* We don't actually know if this is the right device for this
-                 * userid, but we'll record it so we'll at least be consistent
-                 * when the raw device list changes.
-                 */
-                if (rdi.hid.dwVendorId == USB_VENDOR_VALVE &&
-                    rdi.hid.dwProductId == USB_PRODUCT_STEAM_VIRTUAL_GAMEPAD) {
-                    /* Steam encodes the real device in the path */
-                    int realVID = rdi.hid.dwVendorId;
-                    int realPID = rdi.hid.dwProductId;
-                    (void)SDL_sscanf(devName, "\\\\.\\pipe\\HID#VID_045E&PID_028E&IG_00#%x&%x&", &realVID, &realPID);
-                    *pVID = (Uint16)realVID;
-                    *pPID = (Uint16)realPID;
-                    *pVersion = 0;
-                } else {
-                    *pVID = (Uint16)rdi.hid.dwVendorId;
-                    *pPID = (Uint16)rdi.hid.dwProductId;
-                    *pVersion = (Uint16)rdi.hid.dwVersionNumber;
-                }
-                if (s_arrXInputDevicePath[userid]) {
-                    SDL_free(s_arrXInputDevicePath[userid]);
-                }
-                s_arrXInputDevicePath[userid] = SDL_strdup(devName);
-                SDL_free(devices);
-                return;
-            }
-        }
-    }
-    SDL_free(devices);
-#endif /* !__WINRT__ */
+    *pVID = capabilities.VendorId;
+    *pPID = capabilities.ProductId;
+    *pVersion = capabilities.ProductVersion;
```

Comment:
Now that we can match on VID/PID, we should use raw input to look up the device name

Reply:
Can you make these changes so we can cherry pick it as a single commit to SDL2?

Reply:
The VID/PID for all Xbox One controllers appears to be 0x045e/0x02ff, which has the generic controller name, so using raw input to look up the name isn't helpful. Never mind! :)